crates/forge/src/running.rs

```diff
@@ -279,70 +274,29 @@ fn extract_test_case_summary(
 
 fn get_fork_state_reader(
     workspace_root: &Utf8Path,
-    fork_targets: &[ForkTarget],
-    fork_config: &Option<ForkConfig>,
+    fork_config: &Option<ValidatedForkConfig>,
 ) -> Result<Option<ForkStateReader>> {
-    match &fork_config {
-        Some(ForkConfig::Params(url, mut block_id)) => {
+    match fork_config {
+        Some(ValidatedForkConfig { url, mut block_id }) => {
             if let BlockId::Tag(Latest) = block_id {
                 block_id = get_latest_block_number(url)?;
             }
             Ok(Some(ForkStateReader::new(
-                url,
+                url.clone(),
```

Comment:
Why do we need to clone it?

Reply:
It is due to the fact that inside `ForkStateReader::new` we call `HttpTransport::new` of signature `pub fn new(url: impl Into<Url>) -> Self`. `&Url` doesn't implement `Into<Url>` so we would have to clone it anyways. 

Therefore I decide to ensure that `Url` is passed to `ForkStateReader::new`, not `&Url`, so the copy is not hidden somewhere